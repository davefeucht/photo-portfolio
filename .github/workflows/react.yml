# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions

name: React CI/CD

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  build:

    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [18.x]

    steps:
    - name: Checkout code
      uses: actions/checkout@v2

    - name: Install Node.js ${{ matrix.node-version }}
      uses: actions/setup-node@v1
      with:
        node-version: ${{ matrix.node-version }}

    - name: Enable corepack
      run: corepack enable

    - name: Install dependencies
      run: yarn install

    - name: Build project
      run: npm run build --if-present

    - name: Run tests
      run: npm test
