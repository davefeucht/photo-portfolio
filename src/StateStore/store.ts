import { configure, makeAutoObservable } from 'mobx';
import { ImageData, Image, Store, Category, Post, Page } from '../utils/types';

configure({ enforceActions: 'observed' });

class stateStore implements Store {
    menuState = 'closed'
    applicationRoot: HTMLElement = null;
    screenInfo: {
        width: number,
        height: number
    }
    siteInfo: {
        siteName: string,
        siteUrl: string
    }
    visiblePost: {
        postId: number,
        postTitle: string,
        tags: number[],
        tagNames: string[],
        fullImageUrl: string,
        width: number,
        height: number
    };
    categoryList: Category[] = [];
    currentCategoryPosts: Post[] = [];
    currentCategoryData: Category = {
        id: 0,
        count: 0,
        description: '',
        link: '',
        name: '',
        slug: '',
        taxonomy: '',
        parent: 0
    };
    pages: Page[] = [];
    currentPageData: Page = {
        id: 0,
        link: '',
        modified: '',
        slug: '',
        type: '',
        title: {
            rendered: ''
        },
        content: {
            rendered: '',
            protected: false
        },
        author: 0,
        featured_media: 0
    };
    maxItemsPerPage = 10;

    constructor() {
        makeAutoObservable(this);
    }

    setMenuState = (state: string) => {
        this.menuState = state;
    }

    toggleMenuState = () => {
        this.menuState = this.menuState === 'closed' ? 'open' : 'closed';
    }

    setApplicationRoot = (element: HTMLElement) => {
        this.applicationRoot = element;
    }

    setSiteName = (name: string) => {
        this.siteInfo.siteName = name;
    }

    setVisiblePost = (postId: number, postTitle: string) => {
        this.visiblePost.postId = postId;
        this.visiblePost.postTitle = postTitle;
    }

    setVisiblePostImage = (fullImageUrl: string) => {
        this.visiblePost.fullImageUrl = fullImageUrl;
    }

    setVisiblePostTags = (tagNames: string[]) => {
        tagNames.forEach(tagName => {
            this.visiblePost.tagNames.push(tagName);
        });
    }

    clearVisiblePostTagNames = () => {
        this.visiblePost.tagNames.splice(0, this.visiblePost.tagNames.length);
    }

    setCategoryList = (categories: Category[]) => {
        this.categoryList.length = 0;
        categories.forEach((category, index) => {
            this.categoryList[index] = category;
        })
    }

    setCategoryPosts = (posts: Post[]) => {
        this.currentCategoryPosts.length = 0;
        posts.forEach((post, index) => {
            this.currentCategoryPosts[index] = post;
        })
    }

    setCategoryData = (categoryData: Category) => {
        Object.keys(categoryData).forEach((key: keyof Category) => {
            this.currentCategoryData[key] = categoryData[key];
        });
    }

    setThumbnailImageUrl = (imageData: ImageData) => {
        this.currentCategoryPosts[imageData.post_index].thumbnail_image = imageData.image_url;
    }

    setCurrentPost = (postData: Post) => {
        Object.keys(postData).forEach((key: keyof Image) => {
            this.visiblePost[key] = postData[key];
        })
    }

    setPages = (pages: Page[]) => {
        pages.forEach(page => {
            this.pages.push(page);
        });
    }

    setPageData = (pageData: Page) => {
        Object.assign(this.currentPageData, pageData);
    }
}

export default stateStore;
