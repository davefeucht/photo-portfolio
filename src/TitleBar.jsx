/*****************
* TitleBar component implements the title bar of the application
*****************/

import React from "react";


export default class TitleBar extends React.Component {

  render () {
    return (
      <div className="site-header">
        <h1 className="site-title">Through A Pinhole</h1>
      </div>
    );
  }
}