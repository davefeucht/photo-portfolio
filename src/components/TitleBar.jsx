/*****************
* TitleBar component implements the title bar of the application
*****************/

import React from "react";
import StyledTitleBar from "./styledComponents/StyledTitleBar.jsx";
import SiteTitle from "./styledComponents/SiteTitle.jsx";

export default function TitleBar (props) {

    return (
      <StyledTitleBar>
        <SiteTitle>{props.siteName}</SiteTitle>
      </StyledTitleBar>
    );
}