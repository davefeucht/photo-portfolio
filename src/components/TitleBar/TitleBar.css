.title-bar {
  color: #FFFFFF;
  grid-area: header;
  background: rgba(0,0,0,.75); 
  display: grid;
  grid-template-columns: 100%;
  grid-template-rows: 100%;
  align-items: center;
}
.title-bar h1 {
  color: #FFFFFF;
  margin-left: 10px;
}