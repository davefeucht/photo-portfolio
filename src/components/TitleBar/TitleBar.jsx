/*****************
* TitleBar component implements the title bar of the application
*****************/

import React from 'react';
import { observer } from 'mobx-react';
import './TitleBar.css';

const TitleBar = observer(({ siteName }) => {
    return (
      <div className='title-bar'>
        <div>
          <h1>{siteName}</h1>
        </div>
      </div>
    );
});

TitleBar.displayName = 'TitleBar';

export default TitleBar;