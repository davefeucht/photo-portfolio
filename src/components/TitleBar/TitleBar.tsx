/** ***************
* TitleBar component implements the title bar of the application
**************** */

import './TitleBar.css';

import { observer } from 'mobx-react';
import * as React from 'react';
import { useContext, useEffect } from 'react';

import { ApiContext } from '../../utils/ApiContext';
import { API, Store } from '../../utils/types';

interface TitleBarProps {
    stateStore: Store
}

const TitleBar: React.FC<TitleBarProps> = ({ stateStore }) => {
    const { menuState } = stateStore;
    const api = useContext(ApiContext) as API;
    const { getSiteInfo } = api;

    const toggleMenu = () => {
        stateStore.setMenuState(stateStore.menuState === 'closed' ? 'open' : 'closed');
    };

    useEffect(() => {
        getSiteInfo()
            .then(siteName => {
                stateStore.setSiteName(siteName);
                document.title = siteName;
            });
    }, []);

    return (
        <div className="title-bar">
            <div className={`hamburger-button ${menuState}`} onClick={toggleMenu}>
                <img src="./assets/images/hamburger_icon.svg" style={{ border: 0 }} />
            </div>
            <div>
                <h1>{stateStore.siteInfo.siteName}</h1>
            </div>
        </div>
    );
};

TitleBar.displayName = 'TitleBar';

export default observer(TitleBar);
