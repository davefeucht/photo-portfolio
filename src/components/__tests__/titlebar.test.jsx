import { render } from '@testing-library/react';
import React from 'react';

import StateStore from '../../StateStore/store';
import TitleBar from '../TitleBar/TitleBar';

jest.mock('../../utils/Api', () => ({
    getSiteInfo: () => Promise.resolve('Through a Pinhole')
}));

test('Titlebar displays', () => {
    const store = new StateStore();
    const { container } = render(
        <TitleBar stateStore={store} />
    );
    expect(container.firstChild).toMatchSnapshot();
});
