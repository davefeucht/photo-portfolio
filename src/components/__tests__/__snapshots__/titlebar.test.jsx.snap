// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Titlebar displays 1`] = `
<div
  class="title-bar"
>
  <div
    class="hamburger-button closed"
  >
    <img
      src="./assets/images/hamburger_icon.svg"
      style="border: 0px;"
    />
  </div>
  <div>
    <h1>
      Through a Pinhole
    </h1>
  </div>
</div>
`;
